### DISCUSSION TITLE
Build_Profile_Kind

### DISCUSSION THREAD

**Discussion by captain-haddock17** [alire-project/alire#1054] [Q&A]
Hello,
For now `Build_Profile_Kind` values in `*_confg.gpr` has a wording/definition relating to states in a software lifecycle.
"release", "validation", "development".

In my understanding and practices, BUILD setup are intended to define some technical-info/performance details.

So as for other IDE and build tools, I would prefer that the values reflects the (technical) content of the build, i.e "Debug", "Profile", "Release"
```Ada
type Build_Profile_Kind is ("Debug", "Profile", "Release");
```

As a benchmarking info, Apple Xcode defines:
«Build for ... » `Running`, `Testing`, `Profiling`
and compiler switches (& al) have possible `Debug` , `Release` params/options.

These values seems to be buried into 'alr' Ada source code.

Is there any way to redefine this at user/dev level. (by some template ?)

Thanks a lot,
and immense Bravo for all the work already done.

Kind regards,
William

**Comment by Fabien-Chouteau** [comment]
> Is there any way to redefine this at user/dev level. (by some template ?)

The profiles: Release, Validation, Development
and the switches categories: Optimization, Debug_Info, Runtime_Checks, Compile_Checks, Contracts, Style_Checks, Ada_Version
are built-in and users cannot add/remove those.

What users can do is redefine the content of those. For instance:
```toml
[build-switches]
"*".runtime_checks = "none"
release.optimization = ["-O1", "-gnatn"]
```

Since this feature has made it's way into an official release we will need a very strong case to make compatibility breaking changes.
That means we can consider adding a profile and/or a switch category if there is a good argument to back it, but removing or changing is unlikely to happen.

**Reply by captain-haddock17** [reply]
Ah. Ok 
I'll try this’ with feedback . 
Thanks Fabien for the follow-up.
